rust/lance/src/index/vector/ivf.rs

```diff
@@ -139,6 +156,59 @@ impl IVFIndex {
         let batch = part_index.search(&part_query, pre_filter).await?;
         Ok(batch)
     }
+
+    pub(crate) async fn append(
+        &self,
+        dataset: &Dataset,
+        data: impl RecordBatchStream + Unpin,
+        metadata: &IndexMetadata,
+        column: &str,
+    ) -> Result<Uuid> {
+        let new_uuid = Uuid::new_v4();
+        let object_store = dataset.object_store();
+        let index_file = dataset
+            .indices_dir()
+            .child(new_uuid.to_string())
+            .child(INDEX_FILE_NAME);
+        let mut writer = object_store.create(&index_file).await?;
+
+        let pq_index = self
+            .sub_index
+            .as_any()
+            .downcast_ref::<PQIndex>()
+            .ok_or(Error::Index {
+                message: "Only support append to IVF_PQ".to_string(),
+            })?;
+
+        // TODO: merge two IVF implementations.
```

Comment:
I'm not sure I understand this TODO

Reply:
We have half-migrated IVF model to `lance-index`.  I will find sometime next week to move `Ivf` model completely from `lance` crate